addon/v1alpha1/types_addondeploymentconfig.go

```diff
@@ -58,6 +58,31 @@ type AddOnDeploymentConfigSpec struct {
 	// +kubebuilder:validation:MaxLength=63
 	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
 	AgentInstallNamespace string `json:"agentInstallNamespace,omitempty"`
+
+	// Resources required by add-on agents.
+	// If a container matches multiple ContainerResourceRequirements, the last one in the array will
+	// take precedence.
+	// +optional
+	// +listType=map
+	// +listMapKey=containerID
+	AgentResources []ContainerResourceRequirements `json:"agentResources,omitempty"`
+}
+
+// ContainerResourceRequirements defines resources required by one or a group of containers.
+type ContainerResourceRequirements struct {
+	// ContainerID is a unique identifier for an agent container. It consists of three parts: resource kind,
```

Comment:
is this kind or resource type? e.g. should user use Deployment or deployments?

Reply:
It's resource kind, such as Deployment. Of course, it could be resource type as well. Which one is better?

Reply:
Updated the validation pattern to limit the supported resource types.